packages/shared/src/__tests__/telemetry.test.ts

```diff
@@ -170,4 +171,67 @@ describe('TelemetryCollector', () => {
     fetchSpy.mockRestore();
     randomSpy.mockRestore;
   });
+
+  describe('with client-side caching', () => {
+    beforeEach(() => {
+      localStorage.clear();
+    });
+
+    test('sends event when it is not in the cache', () => {
+      const fetchSpy = jest.spyOn(global, 'fetch');
+
+      const collector = new TelemetryCollector({
+        publishableKey: TEST_PK,
+      });
+
+      const event = 'TEST_EVENT';
+
+      collector.record({
+        event,
+        payload: {},
+        clientCache: new TelemetryClientCache({
+          eventKey: `${event}:foo`,
+        }),
+      });
+
+      jest.runAllTimers();
+
+      expect(fetchSpy).toHaveBeenCalled();
+
+      fetchSpy.mockRestore();
+    });
+
+    test('does not send event when it is in the cache', () => {
```

Comment:
Still need to write tests for the following cases:
- When storage isn't available, fallbacks to the event-specific sampling rate
- When there's a `QuotaExceededError`, fallbacks to the event-specific sampling rate